test/e2e/suites/migrate-gitops/suite_test.go

```diff
@@ -179,133 +146,50 @@ var _ = BeforeSuite(func() {
-			HelmBinaryPath:               flagVals.HelmBinaryPath,
-			ChartPath:                    flagVals.ChartPath,
-			CAPIProvidersYAML:            e2e.CapiProviders,
-			Namespace:                    turtlesframework.DefaultRancherTurtlesNamespace,
-			Image:                        fmt.Sprintf("ghcr.io/rancher/turtles-e2e-%s", runtime.GOARCH),
-			Tag:                          "v0.0.1",
-			WaitDeploymentsReadyInterval: e2eConfig.GetIntervals(setupClusterResult.BootstrapClusterProxy.GetName(), "wait-controllers"),
-			AdditionalValues:             map[string]string{},
-		}
-		if flagVals.UseEKS {
-			rtInput.AdditionalValues["rancherTurtles.imagePullSecrets"] = "{regcred}"
-			rtInput.AdditionalValues["rancherTurtles.imagePullPolicy"] = "IfNotPresent"
-		} else {
-			// NOTE: this was the default previously in the chart locally and ok as
-			// we where loading the image into kind manually.
-			rtInput.AdditionalValues["rancherTurtles.imagePullPolicy"] = "Never"
-		}
-
-		rtInput.AdditionalValues["rancherTurtles.features.managementv3-cluster.enabled"] = "false" // disable the default management.cattle.io/v3 controller
-		testenv.DeployRancherTurtles(ctx, rtInput)
+	rtInput := testenv.DeployRancherTurtlesInput{
+		BootstrapClusterProxy:        setupClusterResult.BootstrapClusterProxy,
+		HelmBinaryPath:               flagVals.HelmBinaryPath,
+		ChartPath:                    "https://rancher.github.io/turtles",
+		CAPIProvidersYAML:            e2e.CapiProviders,
+		Namespace:                    turtlesframework.DefaultRancherTurtlesNamespace,
+		Version:                      "v0.6.0",
+		WaitDeploymentsReadyInterval: e2eConfig.GetIntervals(setupClusterResult.BootstrapClusterProxy.GetName(), "wait-controllers"),
+		AdditionalValues:             map[string]string{},
 	}
+	testenv.DeployRancherTurtles(ctx, rtInput)
 
-	if !shortTestOnly() && !localTestOnly() {
-		By("Running full tests, deploying additional infrastructure providers")
-		awsCreds := e2eConfig.GetVariable(e2e.CapaEncodedCredentialsVar)
-		Expect(awsCreds).ToNot(BeEmpty(), "AWS creds required for full test")
-
-		testenv.CAPIOperatorDeployProvider(ctx, testenv.CAPIOperatorDeployProviderInput{
-			BootstrapClusterProxy: setupClusterResult.BootstrapClusterProxy,
-			CAPIProvidersSecretsYAML: [][]byte{
-				e2e.AWSProviderSecret,
-				e2e.AzureIdentitySecret,
-			},
-			CAPIProvidersYAML: e2e.FullProviders,
-			TemplateData: map[string]string{
-				"AWSEncodedCredentials": e2eConfig.GetVariable(e2e.CapaEncodedCredentialsVar),
-			},
-			WaitDeploymentsReadyInterval: e2eConfig.GetIntervals(setupClusterResult.BootstrapClusterProxy.GetName(), "wait-controllers"),
-			WaitForDeployments: []testenv.NamespaceName{
-				{
-					Name:      "capa-controller-manager",
-					Namespace: "capa-system",
-				},
-				{
-					Name:      "capz-controller-manager",
-					Namespace: "capz-system",
-				},
-			},
-		})
-	} else if Label(e2e.LocalTestLabel).MatchesLabelFilter(GinkgoLabelFilter()) {
```

Comment:
I am not sure how vSphere tests are related to downgrade process? Can you please explain why tests here and above are removed completely.

Reply:
They are not 🤷🏻

Reply:
Probably copy paste mistake

Reply:
There is been a discussion of refactoring e2e suite. You can find many of those across different suites. We need a `builder` for suite, which will allow us to construct:
- If it is AKS based execution
- Is VSphere involved
- Is this a local run
- Is it isolated execution using `sslip.io` or not
- Any additional modifications to network, providers, etc.

At this point it is a boilerplate which tends to be duplicated and unfortunately it is mandatory.

Reply:
I opened an issue related to this discussion and previous mentions: https://github.com/rancher/turtles/issues/603